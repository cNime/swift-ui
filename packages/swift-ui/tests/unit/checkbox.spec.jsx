import { mount } from '@vue/test-utils'
import Checkbox from '../../src/components/checkbox/index.vue'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(Checkbox, {
    slots,
    props: { ...values }
  })
}

const checkboxOption = [
  {
    value: 'list1',
    label: 'iPhone X',
    desc:
      '部分地区提供电子普通发票，用户可自行打印，效力等同纸质普通发票，具体以实际出具的发票类型为准。'
  },
  { value: 'list2', label: 'HUAWEI P20' },
  {
    value: 'list3',
    label: 'OPPO Find X',
    desc:
      '部分地区提供电子普通发票，用户可自行打印，效力等同纸质普通发票，具体以实际出具的发票类型为准。',
    disabled: true
  },
  {
    value: 'list4',
    label: 'vivo NEX',
    desc:
      '部分地区提供电子普通发票，用户可自行打印，效力等同纸质普通发票，具体以实际出具的发票类型为准。',
    disabled: true
  }
]

describe('Checkbox Snap', () => {
  it('render initial Checkbox', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Checkbox -- props options', () => {
    const wrapper = factory({ options: checkboxOption })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Checkbox -- props selectedList', () => {
    const wrapper = factory({ options: checkboxOption, selectedList: ['list2'] })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('Checkbox Event', () => {
  it('Checkbox onChange', () => {
    const wrapper = mount(<Checkbox options={checkboxOption} selectedList={['list2']} />)
    expect(wrapper.element).toMatchSnapshot()
    wrapper.find('.at-checkbox .at-checkbox__option').trigger('click')
    expect(wrapper.emitted()).toHaveProperty('change')
  })

  it('Checkbox disabled, onChange not to be called', () => {
    const wrapper = mount(<Checkbox options={checkboxOption} selectedList={['list2']} />)
    wrapper.find('.at-checkbox .at-checkbox__option:nth-child(3)').trigger('click')
    expect(wrapper.emitted().hasOwnProperty('change')).toBe(false)
  })
})
