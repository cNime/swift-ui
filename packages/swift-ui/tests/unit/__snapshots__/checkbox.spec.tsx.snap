// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Event Checkbox onChange 1`] = `
<div
  class="at-checkbox"
>
  
  <div
    class="at-checkbox__option"
  >
    <div
      class="at-checkbox__option-wrap"
    >
      <div
        class="at-checkbox__option-cnt"
      >
        <div
          class="at-checkbox__icon-cnt"
        >
          <span
            class="at-icon at-icon-check"
          />
        </div>
        <div
          class="at-checkbox__title"
        >
          iPhone X
        </div>
      </div>
      <div
        class="at-checkbox__desc"
      >
        Some regions provide electronic ordinary invoices, which users can print by themselves, and the effect is equivalent to paper ordinary invoices. The actual invoice type shall prevail.
      </div>
    </div>
  </div>
  <div
    class="at-checkbox__option at-checkbox__option--selected"
  >
    <div
      class="at-checkbox__option-wrap"
    >
      <div
        class="at-checkbox__option-cnt"
      >
        <div
          class="at-checkbox__icon-cnt"
        >
          <span
            class="at-icon at-icon-check"
          />
        </div>
        <div
          class="at-checkbox__title"
        >
          HUAWEI P20
        </div>
      </div>
      <!---->
    </div>
  </div>
  <div
    class="at-checkbox__option at-checkbox__option--disabled"
  >
    <div
      class="at-checkbox__option-wrap"
    >
      <div
        class="at-checkbox__option-cnt"
      >
        <div
          class="at-checkbox__icon-cnt"
        >
          <span
            class="at-icon at-icon-check"
          />
        </div>
        <div
          class="at-checkbox__title"
        >
          OPPO Find X
        </div>
      </div>
      <div
        class="at-checkbox__desc"
      >
        Some regions provide electronic ordinary invoices, which users can print by themselves, and the effect is equivalent to paper ordinary invoices. The actual invoice type shall prevail.
      </div>
    </div>
  </div>
  <div
    class="at-checkbox__option at-checkbox__option--disabled"
  >
    <div
      class="at-checkbox__option-wrap"
    >
      <div
        class="at-checkbox__option-cnt"
      >
        <div
          class="at-checkbox__icon-cnt"
        >
          <span
            class="at-icon at-icon-check"
          />
        </div>
        <div
          class="at-checkbox__title"
        >
          vivo NEX
        </div>
      </div>
      <div
        class="at-checkbox__desc"
      >
        Some regions provide electronic ordinary invoices, which users can print by themselves, and the effect is equivalent to paper ordinary invoices. The actual invoice type shall prevail.
      </div>
    </div>
  </div>
  
</div>
`;

exports[`Checkbox Snap render Checkbox -- props options 1`] = `
<div
  class="at-checkbox"
>
  
  <div
    class="at-checkbox__option"
  >
    <div
      class="at-checkbox__option-wrap"
    >
      <div
        class="at-checkbox__option-cnt"
      >
        <div
          class="at-checkbox__icon-cnt"
        >
          <span
            class="at-icon at-icon-check"
          />
        </div>
        <div
          class="at-checkbox__title"
        >
          iPhone X
        </div>
      </div>
      <div
        class="at-checkbox__desc"
      >
        Some regions provide electronic ordinary invoices, which users can print by themselves, and the effect is equivalent to paper ordinary invoices. The actual invoice type shall prevail.
      </div>
    </div>
  </div>
  <div
    class="at-checkbox__option"
  >
    <div
      class="at-checkbox__option-wrap"
    >
      <div
        class="at-checkbox__option-cnt"
      >
        <div
          class="at-checkbox__icon-cnt"
        >
          <span
            class="at-icon at-icon-check"
          />
        </div>
        <div
          class="at-checkbox__title"
        >
          HUAWEI P20
        </div>
      </div>
      <!---->
    </div>
  </div>
  <div
    class="at-checkbox__option at-checkbox__option--disabled"
  >
    <div
      class="at-checkbox__option-wrap"
    >
      <div
        class="at-checkbox__option-cnt"
      >
        <div
          class="at-checkbox__icon-cnt"
        >
          <span
            class="at-icon at-icon-check"
          />
        </div>
        <div
          class="at-checkbox__title"
        >
          OPPO Find X
        </div>
      </div>
      <div
        class="at-checkbox__desc"
      >
        Some regions provide electronic ordinary invoices, which users can print by themselves, and the effect is equivalent to paper ordinary invoices. The actual invoice type shall prevail.
      </div>
    </div>
  </div>
  <div
    class="at-checkbox__option at-checkbox__option--disabled"
  >
    <div
      class="at-checkbox__option-wrap"
    >
      <div
        class="at-checkbox__option-cnt"
      >
        <div
          class="at-checkbox__icon-cnt"
        >
          <span
            class="at-icon at-icon-check"
          />
        </div>
        <div
          class="at-checkbox__title"
        >
          vivo NEX
        </div>
      </div>
      <div
        class="at-checkbox__desc"
      >
        Some regions provide electronic ordinary invoices, which users can print by themselves, and the effect is equivalent to paper ordinary invoices. The actual invoice type shall prevail.
      </div>
    </div>
  </div>
  
</div>
`;

exports[`Checkbox Snap render Checkbox -- props selectedList 1`] = `
<div
  class="at-checkbox"
>
  
  <div
    class="at-checkbox__option"
  >
    <div
      class="at-checkbox__option-wrap"
    >
      <div
        class="at-checkbox__option-cnt"
      >
        <div
          class="at-checkbox__icon-cnt"
        >
          <span
            class="at-icon at-icon-check"
          />
        </div>
        <div
          class="at-checkbox__title"
        >
          iPhone X
        </div>
      </div>
      <div
        class="at-checkbox__desc"
      >
        Some regions provide electronic ordinary invoices, which users can print by themselves, and the effect is equivalent to paper ordinary invoices. The actual invoice type shall prevail.
      </div>
    </div>
  </div>
  <div
    class="at-checkbox__option at-checkbox__option--selected"
  >
    <div
      class="at-checkbox__option-wrap"
    >
      <div
        class="at-checkbox__option-cnt"
      >
        <div
          class="at-checkbox__icon-cnt"
        >
          <span
            class="at-icon at-icon-check"
          />
        </div>
        <div
          class="at-checkbox__title"
        >
          HUAWEI P20
        </div>
      </div>
      <!---->
    </div>
  </div>
  <div
    class="at-checkbox__option at-checkbox__option--disabled"
  >
    <div
      class="at-checkbox__option-wrap"
    >
      <div
        class="at-checkbox__option-cnt"
      >
        <div
          class="at-checkbox__icon-cnt"
        >
          <span
            class="at-icon at-icon-check"
          />
        </div>
        <div
          class="at-checkbox__title"
        >
          OPPO Find X
        </div>
      </div>
      <div
        class="at-checkbox__desc"
      >
        Some regions provide electronic ordinary invoices, which users can print by themselves, and the effect is equivalent to paper ordinary invoices. The actual invoice type shall prevail.
      </div>
    </div>
  </div>
  <div
    class="at-checkbox__option at-checkbox__option--disabled"
  >
    <div
      class="at-checkbox__option-wrap"
    >
      <div
        class="at-checkbox__option-cnt"
      >
        <div
          class="at-checkbox__icon-cnt"
        >
          <span
            class="at-icon at-icon-check"
          />
        </div>
        <div
          class="at-checkbox__title"
        >
          vivo NEX
        </div>
      </div>
      <div
        class="at-checkbox__desc"
      >
        Some regions provide electronic ordinary invoices, which users can print by themselves, and the effect is equivalent to paper ordinary invoices. The actual invoice type shall prevail.
      </div>
    </div>
  </div>
  
</div>
`;

exports[`Checkbox Snap render initial Checkbox 1`] = `
<div
  class="at-checkbox"
>
  
  
</div>
`;
